src/pydata_sphinx_theme/theme/pydata_sphinx_theme/sections/sidebar-primary.html

```diff
@@ -10,7 +10,7 @@
 
 {% if sidebars and sidebar_nav_links %}
 <!-- Only show if we have sidebars configured, else just a small margin  -->
-<div class="bd-sidebar-primary col-12 col-md-3 bd-sidebar">
+<div class="bd-sidebar-primary col-6 col-md-3 bd-sidebar">
```

Comment:
Maybe this should be a little bit wide on mobile? Like 60-70% of the screen? That should still leave plenty of space for people to click outside the drawer to close it.